findsecbugs-plugin/src/main/resources/metadata/messages.xml

```diff
@@ -2507,6 +2507,60 @@ DB.withConnection { implicit c =>
     </BugPattern>
     <BugCode abbrev="SECSSQLA">SQL Injection (Scala - Anorm)</BugCode>
 
+    <!-- SQL Injection Vertx -->
+    <BugPattern type="SQL_INJECTION_VERTX">
+        <ShortDescription>Potential SQL Injection with Vert.x Sql Client</ShortDescription>
+        <LongDescription>This use of {3} can be vulnerable to SQL injection (with Vert.x Sql Client)</LongDescription>
+        <Details>
+            <![CDATA[
+<p>
+The input values included in SQL queries need to be passed in safely.
+Bind variables in prepared statements can be used to easily mitigate the risk of SQL injection.
+Vert.x Sql Client API provide a DSL to build query with Java code.
+</p>
+<p>
+    <b>Vulnerable Code:</b><br/>
+    <pre>
+SqlClient.query( "select * from Customer where id=" + inputId ).execute(ar -> ...);
+</pre>
+</p>
+<p>
+    <b>Solution (using Prepared Statements):</b><br/>
+
+    <pre>
+client
+    .preparedQuery( "SELECT * FROM users WHERE id=$1" )
+    .execute(Tuple.of("julien"))
+    .onSuccess(rows -> ...)
+    .onFailure(err -> ...);
+</pre>
+
+    <b>Solution (using OWASP Encoder):</b><br/>
+
+    <pre>
+import org.owasp.esapi.Encoder;
+
+SqlClient
+    .query( "select * from Customer where id=" + Encoder.encodeForSQL(inputId) )
```

Comment:
While FSB eliminates false positive when encoding functions are use, they do not represent the best solution.

For instance, in the code given - if `inputId` is a string - a malicious inputId could be `123 union select * from Users`. The string would not be sanitized because the encoding function assume that the value is placed between single quotes or double quotes. A `union` statement or any form of sub-queries could be introduced as long as no single quote is used. (Ref: [MysqlCodec from ESAPI](https://github.com/ESAPI/esapi-java-legacy/blob/d7bb3a537a76954242db94a6dd2cdb4a88a2856c/src/main/java/org/owasp/esapi/codecs/MySQLCodec.java#L177-L188))
```java
SqlClient
    .query( "select * from Customer where id=" + Encoder.encodeForSQL(inputId) )
```

Bottom line, I would simply remove this second solution. The rest is perfect.